servers/graphql-kotlin-ktor-server/src/main/kotlin/com/expediagroup/graphql/server/ktor/subscriptions/graphqlws/KtorGraphQLWebSocketProtocolHandler.kt

```diff
@@ -0,0 +1,208 @@
+/*
+ * Copyright 2023 Expedia, Inc
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.expediagroup.graphql.server.ktor.subscriptions.graphqlws
+
+import com.expediagroup.graphql.server.execution.GraphQLSubscriptionExecutor
+import com.expediagroup.graphql.server.ktor.subscriptions.KtorGraphQLSubscriptionHandler
+import com.expediagroup.graphql.server.ktor.subscriptions.KtorGraphQLSubscriptionHooks
+import com.expediagroup.graphql.server.ktor.subscriptions.SubscriptionOperationMessage
+import com.expediagroup.graphql.server.types.GraphQLRequest
+import com.fasterxml.jackson.databind.ObjectMapper
+import com.fasterxml.jackson.module.kotlin.convertValue
+import com.fasterxml.jackson.module.kotlin.readValue
+import graphql.GraphQLContext
+import io.ktor.server.websocket.*
+import io.ktor.websocket.*
+import kotlinx.coroutines.Job
+import kotlinx.coroutines.channels.ClosedReceiveChannelException
+import kotlinx.coroutines.flow.map
+import kotlinx.coroutines.flow.onCompletion
+import kotlinx.coroutines.isActive
+import kotlinx.coroutines.launch
+import org.slf4j.LoggerFactory
+import java.util.concurrent.ConcurrentHashMap
+
+/**
+ * Implementation of the `graphql-transport-ws` protocol
+ * https://github.com/enisdenjo/graphql-ws/blob/master/PROTOCOL.md
+ */
+open class KtorGraphQLWebSocketProtocolHandler(
+    private val subscriptionExecutor: GraphQLSubscriptionExecutor,
+    private val objectMapper: ObjectMapper,
+    private val subscriptionHooks: KtorGraphQLSubscriptionHooks,
+) : KtorGraphQLSubscriptionHandler {
+    private val logger = LoggerFactory.getLogger(KtorGraphQLWebSocketProtocolHandler::class.java)
+
+    private val acknowledgeMessage = objectMapper.writeValueAsString(SubscriptionOperationMessage(MessageTypes.GQL_CONNECTION_ACK))
+    private val pongMessage = objectMapper.writeValueAsString(SubscriptionOperationMessage(MessageTypes.GQL_PONG))
+
+    override suspend fun handle(session: WebSocketServerSession) {
+        logger.debug("New client connected")
```

Comment:
log message without context (and there are more below in the class)

Reply:
unfortunately Ktor doesn't provide any ID or something contextual for the `WebSocketSession`. So if there is a new client connected, it doesn't seem obvious what to log (except you generate an artificial ID and log it, but I decided not to do that)

what you will get in the log though, is the coroutine trace, which is good enough I think